File: java/1356.sort-integers-by-the-number-of-1-bits.java
import java.util.Arrays;
import java.util.Comparator;

/*
 * @lc app=leetcode id=1356 lang=java
 *
 * [1356] Sort Integers by The Number of 1 Bits
 */

// @lc code=start
class Solution {
    public int[] sortByBits(int[] arr) {
        Integer[] arrInt = new Integer[arr.length];
        for (int i = 0; i < arr.length; ++i) {
            arrInt[i] = arr[i];
        }
        Arrays.sort(arrInt, Comparator.comparing(i -> Integer.bitCount(i) * (1 << 15) + i));
        for (int i = 0; i < arr.length; ++i)
            arr[i] = arrInt[i];
        return arr;
    }
}
// @lc code=end

File: py/1356.sort-integers-by-the-number-of-1-bits.py
#
# @lc app=leetcode id=1356 lang=python3
#
# [1356] Sort Integers by The Number of 1 Bits
#

# @lc code=start
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        return sorted(arr, key=lambda n: (Solution.countOneBit(n), n))

    @staticmethod
    def countOneBit(n: int) -> int:
        count = 0
        while n:
            if (n & 1):
                count+=1
            n >>= 1
        return count
# @lc code=end
